test/functional/http/post/3.votes.js

```diff
@@ -0,0 +1,464 @@
+'use strict';
+
+var node = require('../../../node');
+var shared = require('../../shared');
+var constants = require('../../../../helpers/constants');
+
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var registerDelegatePromise = require('../../../common/apiHelpers').registerDelegatePromise;
+var waitForBlocksPromise = node.Promise.promisify(node.waitForBlocks);
+
+describe('POST /api/transactions (type 3) votes', function () {
+
+	var badTransactions = [];
+	var goodTransactions = [];
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+
+	var delegateAccount = node.randomAccount();
+	var accountNoFunds = node.randomAccount();
+	var accountMinimalFunds = node.randomAccount();
+	var accountDuplicate = node.randomAccount();
+
+	// Creating two scenarios with two isolated set of accounts
+	var accountMaxVotesPerTransaction = node.randomAccount();
+	var delegatesMaxVotesPerTransaction = [];
+	var accountMaxVotesPerAccount = node.randomAccount();
+	var delegatesMaxVotesPerAccount = [];
+
+	var transaction;
+
```

Comment:
As I mentioned the last time - would be really nice if above the before the section you will write the scenario of those tests as the process is complex